agent/taskresource/credentialspec/credentialspec.go

```diff
@@ -0,0 +1,556 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
Since credentialspec is only going to be supported on windows, can you follow the pattern that firelens has, which has different logic for windows and non windows platform. The reference is [here](https://github.com/aws/amazon-ecs-agent/tree/master/agent/taskresource/firelens)